pkg/pillar/portprober/linuxreachprober.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package portprober
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/utils/netutils"
+	"github.com/tatsushid/go-fastping"
+)
+
+// HostnameAddr is an implementation of net.Addr that can be used to store hostname
+// and optionally also port.
+// Other implementations provided by Golang can only store already resolved IP address.
+type HostnameAddr struct {
+	Hostname string
+	Port     uint16
+}
+
+// Network returns the address's network type.
+func (h *HostnameAddr) Network() string {
+	if h.Port == 0 {
+		return "ip"
+	}
+	// We do not provide UDP-based probing method for now.
+	return "tcp"
+}
+
+// String returns the address in the form "hostname:port".
+func (h *HostnameAddr) String() string {
+	if h.Port == 0 {
+		return h.Hostname
+	}
+	return fmt.Sprintf("%s:%d", h.Hostname, h.Port)
+}
+
+// LinuxReachabilityProber is an implementation of ReachabilityProber
+// for the Linux TCP/IP network stack.
+type LinuxReachabilityProber struct{}
+
+// Probe reachability of <dstAddr> via the given port.
+func (p *LinuxReachabilityProber) Probe(ctx context.Context, portIfName string,
```

Comment:
What about splitting this into `LinuxReachabilityProberICMP` and `LinuxReachabilityProberTCP` - two implementations of `ReachabilityProber`?

Reply:
Makes sense, done.